modules/dnn/src/graph_simplifier.cpp

```diff
@@ -69,8 +69,15 @@ int Subgraph::getInputNodeId(const Ptr<ImportGraphWrapper>& net,
     const int numNodes = net->getNumNodes();
     for (int i = 0; i < numNodes; ++i)
     {
-        if (net->getNodeName(i) == name)
+        Ptr<ImportNodeWrapper> node = net->getNode(i);
+        const int numOutputs = node->getNumOutputs();
+        if (numOutputs == 0 && net->getNodeName(i) == name)
```

Comment:
Can we avoid checking `getNodeName` here? Which tests are failed if we test only `getOutputName` and why?

Reply:
yeah, we can.
Only three tests are failing. 
Reduce_L2
https://github.com/opencv/opencv/blob/8eafddf211ec27091a3096eb50964e04d49b36cb/modules/dnn/test/test_onnx_importer.cpp#L412
Unfused_softmax
https://github.com/opencv/opencv/blob/8eafddf211ec27091a3096eb50964e04d49b36cb/modules/dnn/test/test_onnx_importer.cpp#L438
Div
https://github.com/opencv/opencv/blob/8eafddf211ec27091a3096eb50964e04d49b36cb/modules/dnn/test/test_onnx_importer.cpp#L366
The problem will occur if any node of subgraph that we are trying to match has input from graph's input layer. 
**Solution** 
If we add some layer between our test model node and graph's input. then it will never try to match with graph's input layer.

For example, this will not work
```
class Div(nn.Module):
    def __init__(self):
        super(Div, self).__init__()
    def forward(self, a, b):
        return a/b
a = Variable(torch.randn(1, 3, 2, 4))
b = Variable(torch.randn(1, 3, 2, 4))
model = Div()
save_data_and_model_multy_inputs("division", model, a, b)
```
![Screenshot from 2020-02-23 07-19-50](https://user-images.githubusercontent.com/26292819/75102165-49be6700-560d-11ea-9968-e73e1363736e.png)

But this model will work
```
a = Variable(torch.randn(1, 3, 2, 2))
b = Variable(torch.randn(1, 3, 2, 2))
class DivExp(nn.Module):
    def __init__(self):
        super(DivExp, self).__init__()
    def forward(self, x1, x2):
        flatten1 = x1.flatten().view(1,-1)
        flatten2 = x2.flatten().view(1,-1)
        return flatten1/flatten2
divexp = DivExp()
save_data_and_model_multy_inputs("divexp", divexp, a , b)
```
![Screenshot from 2020-02-23 07-20-28](https://user-images.githubusercontent.com/26292819/75102170-5347cf00-560d-11ea-864b-e8617bae5e9d.png)

Reply:
@dkurt should I make some changes in these tests [`Reduce_L2`, `Div `and `Softamx_unfused`] ?